model-optimizer/mo/ops/memoryoffset.py

```diff
@@ -30,17 +30,18 @@ def __init__(self, graph: Graph, attrs: dict):
             'pair_name': None,
             'splitted': False,
             'has_default': False,
-            'infer': __class__.infer,
+            'infer': self.infer,
             'in_ports_count': 1,
             'out_ports_count': 1,
         }, attrs)
 
-
     @staticmethod
     def infer(node: Node):
         if node.has_valid('element_size'):
             # element_size should be set by Kaldi loader or by MemoryOffsetAdjustment
-            node.out_port(0).data.set_shape([1, node['element_size']])
+            # batch added to Kaldi by MO and is not changing in graph
+            batch = node.graph.get_op_nodes(op="Parameter")[0].shape[0]
```

Comment:
Getting all Parameter nodes for each memory node will be time-consuming, maybe there is another way to do this?

Reply:
This is very unstable approach. What if the model have multiple inputs and some of these inputs do not have "batch" dimension?

Reply:
Model can't have inputs without batch or with another batch, we add it in MO for each input because Kaldi have only 1 dimension. I don't like this approach too, but I have no other ways that don't create loops (from this point of view the approach is stable).

Reply:
I think I can try to move adding broadcasted constant for ReadValue to separate transformation and call it once for all ReadValue nodes, it allows to search Parameter only once and check that we do it only for Kaldi (in case in future we will have ReadValue in some other frameworks). Additionally we can check that all Parameters have the same batch.

Reply:
Sounds great! Let's do this!

Reply:
added transformation with test